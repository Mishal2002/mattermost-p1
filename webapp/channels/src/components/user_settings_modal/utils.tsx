// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {defineMessages, useIntl} from 'react-intl';

import {
    BellOutlineIcon,
    DockLeftIcon,
    ForumOutlineIcon,
    GlobeIcon,
    PaletteOutlineIcon,
    TuneIcon,
} from '@mattermost/compass-icons/components';

import type {Tab} from 'components/widgets/modals/components/modal_sidebar';

import {t} from 'utils/i18n';

export const holders = defineMessages({
    profile: {
        id: 'user.settings.modal.profile',
        defaultMessage: 'Profile',
    },
    security: {
        id: 'user.settings.modal.security',
        defaultMessage: 'Security',
    },
    notifications: {
        id: 'user.settings.modal.notifications',
        defaultMessage: 'Notifications',
    },
    themes: {
        id: 'user.settings.modal.themes',
        defaultMessage: 'Themes',
    },
    media: {
        id: 'user.settings.modal.media',
        defaultMessage: 'Messages & media',
    },
    display: {
        id: 'user.settings.modal.display',
        defaultMessage: 'Display',
    },
    language: {
        id: 'user.settings.modal.language',
        defaultMessage: 'Language & time',
    },
    sidebar: {
        id: 'user.settings.modal.sidebar',
        defaultMessage: 'Sidebar',
    },
    advanced: {
        id: 'user.settings.modal.advanced',
        defaultMessage: 'Advanced',
    },
    checkEmail: {
        id: 'user.settings.general.checkEmail',
        defaultMessage: 'Check your email at {email} to verify the address. Cannot find the email?',
    },
    confirmTitle: {
        id: 'user.settings.modal.confirmTitle',
        defaultMessage: 'Discard Changes?',
    },
    confirmMsg: {
        id: 'user.settings.modal.confirmMsg',
        defaultMessage: 'You have unsaved changes, are you sure you want to discard them?',
    },
    confirmBtns: {
        id: 'user.settings.modal.confirmBtns',
        defaultMessage: 'Yes, Discard',
    },
});
export function useUserSettingsTabs(): Tab[] {
    const color = 'currentcolor';
    const {formatMessage} = useIntl();
    return [
        {
            icon: (
                <BellOutlineIcon
                    size={18}
                    color={color}
                />),
            name: 'notifications',
            uiName: formatMessage(holders.notifications),
        },
        {
            icon: (
                <PaletteOutlineIcon
                    size={18}
                    color={color}
                />),
            name: 'themes',
            uiName: formatMessage(holders.themes),
        },
        {
            icon: (
                <ForumOutlineIcon
                    size={18}
                    color={color}
                />),
            name: 'media',
            uiName: formatMessage(holders.media),
        },
        {
            icon: (
                <GlobeIcon
                    size={18}
                    color={color}
                />),
            name: 'language',
            uiName: formatMessage(holders.language),
        },
        {
            icon: (
                <DockLeftIcon
                    size={18}
                    color={color}
                />),
            name: 'sidebar',
            uiName: formatMessage(holders.sidebar),
        },
        {
            icon: (
                <TuneIcon
                    size={18}
                    color={color}
                />),
    name: 'advanced',
            uiName: formatMessage(holders.advanced),
        },
    ];
}
