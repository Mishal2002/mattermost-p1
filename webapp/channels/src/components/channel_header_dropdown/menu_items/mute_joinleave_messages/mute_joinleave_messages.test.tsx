// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Menu from 'components/widgets/menu/menu';

import {TestHelper} from 'utils/test_helper';

import MuteJoinLeaveMessages from './mute_joinleave_messages';

describe('components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages', () => {
    const baseProps = {
        channel: TestHelper.getChannelMock({
            id: 'channel_id',
            type: 'O',
        }),
        actions: {
            patchChannel: jest.fn(),
        },
    };

    it('should match snapshot', () => {
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should read "show join/leave messages"', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                exclude_post_types: ['system_join_channel', 'system_leave_channel'],
            },
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should read "hide join/leave messages"', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                exclude_post_types: [],
            },
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should run patchChannel function on click to set as muted', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                exclude_post_types: [],
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            exclude_post_types: ['system_join_channel', 'system_leave_channel'],
        });
    });
    it('should run patchChannel function on click to set as unmuted', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                exclude_post_types: ['system_join_channel', 'system_leave_channel'],
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            exclude_post_types: [],
        });
    });
    it('should run patchChannel function on click to set as unmuted and persist other exclude types', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                exclude_post_types: ['system_join_channel', 'system_leave_channel', 'another_one'],
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            exclude_post_types: ['another_one'],
        });
    });
});
