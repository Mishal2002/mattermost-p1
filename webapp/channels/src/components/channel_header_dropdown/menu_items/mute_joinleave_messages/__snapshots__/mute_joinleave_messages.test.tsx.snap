// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should match snapshot 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text="Hide Join/Leave Messages"
/>
`;

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should read "hide join/leave messages" 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text="Hide Join/Leave Messages"
/>
`;

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should read "show join/leave messages" 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text="Show Join/Leave Messages"
/>
`;
